src/ProposalTypesConfigurator.sol

```diff
@@ -216,21 +222,53 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
                     endIdx = endIdx + validScope.parameters[j].length;
 
                     bytes32 param = bytes32(proposedTx[startIdx:endIdx]);
-                    if (validScope.comparators[j] == Comparators.EQUAL) {
-                        bytes32 scopedParam = bytes32(validScope.parameters[j]);
-                        if (scopedParam != param) revert InvalidParamNotEqual();
+
+                    if (validScope.types[j] == SupportedTypes.UINT8) {
+                        validator.validate_uint8(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
+
+                    if (validScope.types[j] == SupportedTypes.UINT16) {
+                        validator.validate_uint16(
+                            proposedTx[startIdx:endIdx], validScope.parameters[j], validScope.comparators[j]
+                        );
+                    }
```

Comment:
we should be using if else, and then at the end have an else that reverts if the type is not found.

Reply:
for example: https://github.com/vertex-protocol/vertex-contracts/blob/main/contracts/Endpoint.sol#L432-L510